lucille-parent/pom.xml

```diff
@@ -48,9 +48,9 @@
     <solr.version>9.5.0</solr.version>
     <opensearch-rest-client.version>1.2.0</opensearch-rest-client.version>
     <opensearch.version>0.1.0</opensearch.version>
-    <opensearch-java.version>2.6.0</opensearch-java.version>
-    <httpcore5.version>5.1.5</httpcore5.version>
-    <httpclient5.version>5.1.4</httpclient5.version>
+    <opensearch-java.version>2.8.2</opensearch-java.version>
+    <httpcore5.version>5.2.4</httpcore5.version>
+    <httpclient5.version>5.3</httpclient5.version>
```

Comment:
The opensearch docs [here](https://opensearch.org/docs/latest/clients/java/#installing-the-client-using-apache-httpclient-5-transport) list the opensearch java client version 2.8.1 with httpclient5 version 5.2.1.

I see that this PR has 2.8.2 with 5.3.

The latest opensearch java client seems to be 2.11.1 
https://github.com/opensearch-project/opensearch-java/tree/v2.11.1
https://mvnrepository.com/artifact/org.opensearch.client/opensearch-java/2.11.1

So we're bumping the version to higher than what's in the docs but lower than the latest released version. Looks like the OS docs might be behind? Should we try upgrading all the way to 2.11.1?

Reply:
We can try going to the latest client version... I'm more curious, why are we/do we have to override httpcore4 and httpclient5 versions here?  I'd hope they would just come in by default from the opensearch-java client..

Reply:
Looks like they're not transitive dependencies of opensearch-java-client. Tried commenting them out of lucille-core's pom.xml and got compilation errors like "package org.apache.hc.client5.http.auth does not exist"
(Even if they we're transitive dependencies, when we move Opensearch to a separate module, we'd still need these in core.)

Reply:
I've tested my affected pipeline and forced an upgrade in my project and it works with these newer versions just fine. For reference, the project that refers to lucille I was able to make the following changes:

```
  <properties>
    ...
    <opensearch-java.version>2.11.1</opensearch-java.version>
    <httpcore5.version>5.2.4</httpcore5.version>
    <httpclient5.version>5.3.1</httpclient5.version>
    ...
  </properties>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>org.opensearch.client</groupId>
        <artifactId>opensearch-java</artifactId>
        <version>${opensearch-java.version}</version>
      </dependency>
      <dependency>
        <groupId>org.apache.httpcomponents.client5</groupId>
        <artifactId>httpclient5</artifactId>
        <version>${httpclient5.version}</version>
      </dependency>
      <dependency>
        <groupId>org.apache.httpcomponents.core5</groupId>
        <artifactId>httpcore5</artifactId>
        <version>${httpcore5.version}</version>
      </dependency>
    </dependencies>
  </dependencyManagement>
```

And this was enough to override so only these versions were in the target/libs, not the older ones. An approach like this seems to allow us to try out newer libraries despite release that refers to older libraries.